src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -12,11 +12,40 @@
 using System.Collections.ObjectModel;
 using System.Linq;
 using System.Net;
+using System.Threading.Tasks;
 
 namespace Neo.Network.P2P
 {
     partial class RemoteNode
     {
+        private class TransactionRouter : UntypedActor
```

Comment:
Why do we need this?

Reply:
This is used for actor pool. Creating a actor pool means creating multiple objects, so the best solution is to seperate corresponding logic into a class as small as possible.

Reply:
If it is a pool, then it should be static. I don't think we should create a pool for each remote node. And I think maybe we don't need a pool. We can just process the transactions in `RemoteNode` actor.

Reply:
This PR aims to enhance TPS, and can potentially increase TPS by 3 ~ 4 times.

By experiment it is clear that currently transaction witness verification is the bottleneck of whole system. If we don't speed up transaction witness verification, TPS will be severely limited no matter what we optimize in other aspects.

Currently we come up with 3 kinds of optimization methods:

(1) Starting new tasks for incoming transaction payloads, which is the original logic of this PR

(2) Keep a static thread pool for incoming transaction payloads, which is by @Shargon and described in Qiao-Jin#39

(3) Establish an actor pool for each remotenode handling transaction verification in parallel

All three methods aim to run transaction verification in parallel. By experiment their effect is (1) > (2) = (3)

If we don't need to enhance TPS and enhance system robustness, maybe we can keep elder logic for remotenode. If we do, the bottleneck in transaction witness verification must be solved before other optimizations.

Reply:
An actor pool is a thread pool. My idea is to share an actor pool with all the remote nodes. Multiple actor pools won't enhance TPS.

Reply:
Have changed the logic to add a public transactionrouter, please have a look

Reply:
If so, why not use option(1) directly, all of the three options are based on thread pool, but option1 has the best performance.

_Task Parallel Library(TPL) which mixes features of Thread(notification) & ThreadPool (auto management of threads). Task uses a TaskScheduler which schedules the job work and works on ThreadPool. [link](https://codepattern.net/Blog/post/thread-vs-threadpool-vs-task-in-dot-net)_

_BUT be aware that:
**I/O Bound**: For I/O bound operations (database calls, read/write files, APIs calls, etc) never use normal tasks, use LongRunning tasks or threads if you need to, but not normal tasks. Because it would lead you to a thread pool with a few threads busy and a lot of another tasks waiting for its turn to take the pool.
**CPU Bound**: For CPU bound operations just use the normal tasks and be happy. [link](https://stackoverflow.com/questions/1774670/c-sharp-threadpool-vs-tasks)_